src/modules/roc_netio/target_uv/roc_netio/basic_port.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_netio/target_uv/roc_netio/basic_port.h
+//! @brief basic network port.
+
+#ifndef ROC_NETIO_BASIC_PORT_H_
+#define ROC_NETIO_BASIC_PORT_H_
+
+#include "roc_core/iallocator.h"
+#include "roc_core/list_node.h"
+#include "roc_core/refcnt.h"
+
+#include "roc_packet/address.h"
+
+namespace roc {
+namespace netio {
+
+//! Basic port interface.
+class BasicPort : public core::RefCnt<BasicPort>, public core::ListNode {
+public:
+    //! Initialize.
+    explicit BasicPort(core::IAllocator&);
+
+    //! Destroy.
+    virtual ~BasicPort();
+
+    //! Get bind address.
+    virtual const packet::Address& address() const = 0;
+
+    //! Open port.
+    //! @remarks
+    //!  Should be called from the event loop thread.
+    virtual bool open(packet::Address& bind_address) = 0;
```

Comment:
It's worth to mention in the comment that open() modifies `@p bind_address` if the port is zero.

Reply:
done